torchvision/datasets/voc.py

```diff
@@ -0,0 +1,264 @@
+import os
+import sys
+import tarfile
+import torch.utils.data as data
+if sys.version_info[0] == 2:
+    import xml.etree.cElementTree as ET
+else:
+    import xml.etree.ElementTree as ET
+
+from PIL import Image
+from .utils import download_url, check_integrity
+
+VOC_CLASSES = [
+    'background', 'aeroplane', 'bicycle', 'bird', 'boat', 'bottle', 'bus',
+    'car', 'cat', 'chair', 'cow', 'diningtable', 'dog', 'horse', 'motorbike',
+    'person', 'pottedplant', 'sheep', 'sofa', 'train', 'tvmonitor'
+]
+DATASET_YEAR_DICT = {
+    '2012': [
+        'http://host.robots.ox.ac.uk/pascal/VOC/voc2012/VOCtrainval_11-May-2012.tar',
+        'VOCtrainval_11-May-2012.tar', '6cd6e144f989b92b3379bac3b3de84fd',
+        ' VOCdevkit/VOC2012'
+    ],
+    '2011': [
+        'http://host.robots.ox.ac.uk/pascal/VOC/voc2011/VOCtrainval_25-May-2011.tar',
+        'VOCtrainval_25-May-2011.tar', '6c3384ef61512963050cb5d687e5bf1e',
+        'TrainVal/VOCdevkit/VOC2011'
+    ],
+    '2010': [
+        'http://host.robots.ox.ac.uk/pascal/VOC/voc2010/VOCtrainval_03-May-2010.tar',
+        'VOCtrainval_03-May-2010.tar', 'da459979d0c395079b5c75ee67908abb',
+        'VOCdevkit/VOC2010'
+    ],
+    '2009': [
+        'http://host.robots.ox.ac.uk/pascal/VOC/voc2009/VOCtrainval_11-May-2009.tar',
+        'VOCtrainval_11-May-2009.tar', '59065e4b188729180974ef6572f6a212',
+        'VOCdevkit/VOC2009'
+    ],
+    '2008': [
+        'http://host.robots.ox.ac.uk/pascal/VOC/voc2008/VOCtrainval_14-Jul-2008.tar',
+        'VOCtrainval_14-Jul-2008.tar', '2629fa636546599198acfcfbfcf1904a',
+        'VOCdevkit/VOC2008'
+    ],
+    '2007': [
+        'http://host.robots.ox.ac.uk/pascal/VOC/voc2007/VOCtrainval_06-Nov-2007.tar',
+        'VOCtrainval_06-Nov-2007.tar', 'c52e279531787c972589f7e41ab4ae64',
+        'VOCdevkit/VOC2007'
+    ]
+}
+
+
+class VOCSegmentation(data.Dataset):
+    """`Pascal VOC <http://host.robots.ox.ac.uk/pascal/VOC/>`_ Segmentation Dataset.
+
+    Args:
+        root (string): Root directory of the VOC Dataset.
+        year (string, optional): The dataset year, supports years 2007 to 2012.
+        image_set (string, optional): Select the image_set to use, ``train, trainval or val``
```

Comment:
I would wrap `train`, `trainval`, and `val` in double backticks separately.

Reply:
Good catch! And it'd be helpful if you tried some years (2011,2018) I did the test for all but just in case.